addons/promote/model-java/src/main/java/org/commonjava/indy/promote/model/AbstractPromoteRequest.java

```diff
@@ -0,0 +1,37 @@
+package org.commonjava.indy.promote.model;
+
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * Created by ruhan on 12/5/18.
+ */
+public abstract class AbstractPromoteRequest<T extends PromoteRequest> implements PromoteRequest
+{
+    @ApiModelProperty( value="Asynchronous call. A callback url is needed when it is true." )
+    protected boolean async;
+
+    @ApiModelProperty( value="Callback Url which is used to send the promotion result." )
+    protected String callbackUrl;
```

Comment:
Instead of string only I would let a caller specify how should the callback look like.
https://github.com/project-ncl/causeway/blob/master/rest/src/main/java/org/jboss/pnc/causeway/rest/CallbackTarget.java

Reply:
Thanks. I add the same CallbackTarget to Indy.